### PR TITLE
Image Customizer: Support string mountPoint

### PR DESCRIPTION
**Pull request by Adub17030MS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Currently the API looks like:
```
storage:
  filesystems:
  - deviceId: boot
    type: ext4
    mountPoint:
      path: /boot
```
This PR adds support for:
```
storage:
  filesystems:
    mountPoint: /boot
```
for when the user doesn't want to add mount options.
###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Validated with local testing

### CODE REVIEW

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/imagecustomizerapi/mountpoint.go [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
+		// Treat scalar value as the Path directly.
+		p.Path = value.Value
+		return nil
+	}
+
+	// Otherwise, decode as a full MountPoint struct.
+	type Alias MountPoint
+	var mp Alias
+	if err := value.Decode(&mp); err != nil {
+		return fmt.Errorf("failed to parse MountPoint struct: %w", err)
```
[the remark this answers is not in the document]
Comment (Adub17030MS) [reply toolkit/tools/imagecustomizerapi/mountpoint.go]: Updated!

File: toolkit/tools/imagecustomizerapi/mountpoint.go [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
 	"fmt"
+
+	"gopkg.in/yaml.v3"
 )
 
 // MountPoint holds the mounting information for each partition.
 type MountPoint struct {
 	// The ID type to use for the source in the /etc/fstab file.
-	IdType MountIdentifierType `yaml:"idType"`
+	IdType MountIdentifierType `yaml:"idType,omitempty"`
```
[the remark this answers is not in the document]
Comment (Adub17030MS) [reply toolkit/tools/imagecustomizerapi/mountpoint.go]: Removed.

File: toolkit/tools/imagecustomizerapi/mountpoint.go [hunk 3]
```diff
@@ -1,20 +1,38 @@
...
+func (p *MountPoint) UnmarshalYAML(value *yaml.Node) error {
+	// Check if the node is a scalar (i.e., single path string).
+	if value.Kind == yaml.ScalarNode {
+		// Treat scalar value as the Path directly.
+		p.Path = value.Value
+		return nil
+	}
+
+	// Otherwise, decode as a full MountPoint struct.
+	type Alias MountPoint
```
[the remark this answers is not in the document]
Comment (Adub17030MS) [reply toolkit/tools/imagecustomizerapi/mountpoint.go]: Updated!

File: toolkit/tools/imagecustomizerapi/mountpoint.go [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
+	var mp Alias
+	if err := value.Decode(&mp); err != nil {
```
[the remark this answers is not in the document]
Comment (Adub17030MS) [reply toolkit/tools/imagecustomizerapi/mountpoint.go]: Updated to use this implementation.

File: toolkit/tools/imagecustomizer/docs/configuration.md [hunk 5]
```diff
@@ -1010,6 +1010,27 @@ The meaning of this value depends on the type property.
 
 ## mountPoint type
 
+#### Usage
```
Comment (cwize1) [inline toolkit/tools/imagecustomizer/docs/configuration.md:1013]: nit: Remove the `#### Usage` header.

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/imagecustomizerapi/mountpoint.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (cwize1) [reply toolkit/tools/imagecustomizerapi/mountpoint.go]: Yes, looks good.

File: toolkit/tools/imagecustomizer/docs/configuration.md (hunk 5 above)
Comment (Adub17030MS) [reply toolkit/tools/imagecustomizer/docs/configuration.md]: Removed!

**Review by cwize1 — APPROVED** [review]

**Pull request merged**